src/directives/KeyStroke/index.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+import { onKeyStroke } from '@vueuse/core'
+
+const disableKeyboardShortcuts = window.OCP?.Accessibility?.disableKeyboardShortcuts?.()
+
+/**
+ * Check if event target (active element) is interactive and should not trigger the directive callback
+ * @param {KeyboardEvent} event keyboard event
+ * @return {boolean} whether it should prevent callback
+ */
+function shouldPreventCallback(event) {
+	/** Abort the directive if active element is an input, textarea or contenteditable */
+	if (event.target instanceof HTMLInputElement
+		/**
+		 * TODO discuss if we should abort on another interactive elements
+		 * || event.target instanceof HTMLButtonElement
+		 * || event.target instanceof HTMLAnchorElement
+		 * || event.target instanceof HTMLSelectElement
+		 * || (event.target as HTMLElement)?.hasAttribute('tabindex')
+		 */
+		|| event.target instanceof HTMLTextAreaElement
+		|| event.target?.isContentEditable) {
+		return true
+	}
+	/** Abort the directive if any modal/dialog opened */
+	return document.getElementsByClassName('modal-mask').length !== 0
+}
+
+const eventHandler = (callback, modifiers) => (event) => {
+	if (!!modifiers.ctrl !== event.ctrlKey) {
+		// Ctrl is required and not pressed, or the opposite
+	} else if (disableKeyboardShortcuts) {
+		// Keyboard shortcuts are disabled
+	} else if (shouldPreventCallback(event)) {
+		// Keyboard shortcuts are disabled, because active element assumes input
+	} else {
+		if (modifiers.prevent) {
+			event.preventDefault()
+		}
+		if (modifiers.stop) {
+			event.stopPropagation()
+		}
+		callback(event)
+	}
+}
+
+/**
+ * Wrapper for https://vueuse.org/core/onKeyStroke/
+ * Available modifiers:
+ * - .ctrl: Ctrl key is required
+ * - .prevent: Prevent default action
+ * - .stop: Stop propagation
+ * - .push: Trigger on keydown and keyup
+ * - .press: Trigger on keydown, don't dedupe events
+ */
+const directive = {
+	bind(el, binding) {
+		const keys = binding.arg?.replace('space', ' ').split(',') ?? true
+		onKeyStroke(keys, eventHandler(binding.value, binding.modifiers), {
+			eventName: 'keydown',
+			dedupe: !binding.modifiers.press,
+			target: disableKeyboardShortcuts ? null : document,
+		})
+		if (binding.modifiers.push) {
+			onKeyStroke(keys, eventHandler(binding.value, binding.modifiers), {
+				eventName: 'keyup',
+				target: disableKeyboardShortcuts ? null : document,
+			})
+		}
+	},
+}
```

Comment:
I'd be against implementing this as a directive, because it is technically not a directive.

By design, directive defines node's behavior. Like `v-show` defines its styles, `v-model` it's value property and updating event and etc.

This directive is bind to an element, but has no relation with the element. It doesn't use the `el` argument. When it's used in a component, moving directive from one node to another change nothing.

It even doesn't have `unbind`, so when the element with the directive is destroyed, the directive is still working until the component is unmounted.

IMO, it either can be implemented as a composable (because it is a wrapper around `onKeyStroke` composable). Or it can be actually bind to an element, for example, if short key means "click on this element" and not "call any handler".

Reply:
Had it rewritten to a simple library function wrapper (with custom modifiers basically).

Tested with `script setup`, and options API hooks. Works correctly, adds and removes listeners correctly